### PR TITLE
KEP-2681: Docs update for Alpha PodHostIPs

### PR DESCRIPTION
**Pull request by wzshiming**
ref: https://github.com/kubernetes/enhancements/issues/2681

### CODE REVIEW

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 1]
```diff
@@ -651,6 +652,7 @@ Each feature gate is designed for enabling/disabling a specific feature:
 - `PodAndContainerStatsFromCRI`: Configure the kubelet to gather container and pod stats from the CRI container runtime rather than gathering them from cAdvisor.
   As of 1.26, this also includes gathering metrics from CRI and emitting them over `/metrics/cadvisor` (rather than having cAdvisor emit them directly).
 - `PodDisruptionConditions`: Enables support for appending a dedicated pod condition indicating that the pod is being deleted due to a disruption.
+- `PodHostIPs`: Enables the `status.hostIPs` field for pods and downward API to expose host IPs.
```
Comment (sftim) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:655]:
```suggestion
- `PodHostIPs`: Enable the `status.hostIPs` field for pods and the {{< glossary_tooltip term_id="downward-api" text="downward API" >}}. The field lets you expose host IP addresses to workloads.
```

**Review by sftim — CHANGES_REQUESTED** [review]
I assume that we'll also need an update to https://kubernetes.io/docs/concepts/workloads/pods/downward-api/#downwardapi-fieldRef to mention the new field and the feature gate that controls it.

Please add that.

**Comment by wzshiming** [comment]
Done

**Comment by sftim** [comment]
> Done

I still don't see an update to the downward API docs.

**Comment by wzshiming** [comment]
Updated, sorry, I didn't notice it before.

File: content/en/docs/concepts/workloads/pods/downward-api.md [hunk 2]
```diff
@@ -75,9 +75,15 @@ The following information is available through environment variables
 `status.hostIP`
 : the primary IP address of the node to which the Pod is assigned
 
+`status.hostIPs`
+: the IP addresses is a dual-stack version of `status.hostIP`, the first is always the same as `status.hostIP`
```
Comment (sftim) [inline content/en/docs/concepts/workloads/pods/downward-api.md:79]:
```suggestion
: the IP addresses is a dual-stack version of `status.hostIP`, the first is always the same as `status.hostIP`. The field is available if you enable the `PodHostIPs` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/).
```

File: content/en/docs/concepts/workloads/pods/downward-api.md
Comment (sftim) [inline content/en/docs/concepts/workloads/pods/downward-api.md:1]: We might need to change
```diff
-For most Pod-level fields, you can provide them to a container either as
+For some Pod-level fields, you can provide them to a container either as
an environment variable or using a `downwardAPI` volume. The fields available
via either mechanism are:
```

**Comment by wzshiming** [comment]
done

File: content/en/docs/reference/kubernetes-api/workload-resources/pod-v1.md
Comment (sftim) [inline content/en/docs/reference/kubernetes-api/workload-resources/pod-v1.md:1]: Don't update this file. We'll automatically update it during the release process.

File: content/en/docs/reference/kubernetes-api/cluster-resources/node-v1.md
Comment (sftim) [inline content/en/docs/reference/kubernetes-api/cluster-resources/node-v1.md:1]: Don't update this file. We'll automatically update it during the release process.

**Comment by sftim** [comment]
I think this is ready for a tech review.

File: content/en/docs/concepts/workloads/pods/downward-api.md [hunk 3]
```diff
@@ -75,9 +75,15 @@ The following information is available through environment variables
 `status.hostIP`
 : the primary IP address of the node to which the Pod is assigned
 
+`status.hostIPs`
+: the IP addresses is a dual-stack version of `status.hostIP`, the first is always the same as `status.hostIP`. The field is available if you enable the `PodHostIPs` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/).
```
Comment (tengqm) [inline content/en/docs/concepts/workloads/pods/downward-api.md:79]: please manually wrap the long line as we do for other paragraphs.

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 4]
```diff
@@ -655,6 +656,7 @@ Each feature gate is designed for enabling/disabling a specific feature:
 - `PodAndContainerStatsFromCRI`: Configure the kubelet to gather container and pod stats from the CRI container runtime rather than gathering them from cAdvisor.
   As of 1.26, this also includes gathering metrics from CRI and emitting them over `/metrics/cadvisor` (rather than having cAdvisor emit them directly).
 - `PodDisruptionConditions`: Enables support for appending a dedicated pod condition indicating that the pod is being deleted due to a disruption.
+- `PodHostIPs`: Enable the `status.hostIPs` field for pods and the {{< glossary_tooltip term_id="downward-api" text="downward API" >}}. The field lets you expose host IP addresses to workloads.
```
Comment (tengqm) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:659]: Please wrap the long line.

**Review by sftim — COMMENTED** [review]
Thanks

/lgtm
/approve

**Pull request merged**